engine/src/gldrv/gl_light_state.cpp

```diff
@@ -338,11 +355,11 @@ void gfx_light::ResetProperties( const enum LIGHT_TARGET light_targ, const GFXCo
 {
     bool changed = false;
     if ( LocalLight() ) {
-        GFXLight t;
-        memcpy( &t, this, sizeof (GFXLight) );
-        t.SetProperties( light_targ, color );
-        changed = RemoveFromTable( false, t );
-        memcpy( this, &t, sizeof (GFXLight) );
+        //GFXLight t;
+        //memcpy( &t, this, sizeof (GFXLight) );
+        this->SetProperties( light_targ, color );
+        changed = RemoveFromTable( false, *this );
+        //memcpy( this, &t, sizeof (GFXLight) );
```

Comment:
Not sure about this part. Everything else looks good to me.

Reply:
@stephengtuggy I will look at it a little more closely now. Thanks for pointing it out. I thought it was just pointlessly copied back and fourth, but now this "changed" function is intriguing me.

Reply:
@stephengtuggy I have reverted the code to previous behavior. However, I have implemented a copy construct to replace the memsets.

Reply:
@evertvorster Cool, thanks. This looks correct now.